### DISCUSSION TITLE
Right alignment of text

### DISCUSSION THREAD

**Discussion by sourcebox** [emilk/egui#701] [Q&A] [answered]
What is the best practice to right align text within a container? Use a grid?

E.g. I have a footer made up of the following code:

```rust
egui::TopBottomPanel::bottom("bottom_panel").show(ctx, |ui| {
    ui.horizontal(|ui| {
        ui.label(format!("v{}", &env!("CARGO_PKG_VERSION")));
        egui::warn_if_debug_build(ui);
        ui.hyperlink_to("Project homepage", env!("CARGO_PKG_HOMEPAGE"));
    });
});
```
I want the link to be right aligned.

**Comment by emilk** [comment, marked as answer]
Instead of `ui.horizontal`, use:

``` rust
ui.with_layout(egui::Layout::right_to_left(), |ui| {
    …
})
```

**Reply by Maikel1963** [reply]
I have the same issue, but in the center panel - rendering a vec of some links with a function. "with layout" does not really help, as it expands the first hyperlink to fill the remaining part of the window. All the others have correct size.
Is there a way to avoid this?
(using version 0.17)

**Reply by nixpulvis** [reply]
I don't understand why this is the recommended answer since it changes the tab ordering to be incorrect.

Perhaps there should be an option to reverse the tab order within a RtL layout?
